packages/dashboard/src/components/widgets/primitives/input/index.tsx

```diff
@@ -1,22 +1,32 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
 import { Button, Select, SelectProps } from '@cloudscape-design/components';
 import SpaceBetween from '@cloudscape-design/components/space-between';
 import { InputWidget as InputWidgetType } from '../../../../types';
 
 export type InputWidgetProps = InputWidgetType & { readOnly: boolean };
 
 const Input: React.FC<InputWidgetProps> = ({ readOnly, ...widget }) => {
-  const [selectedOption, setSelectedOption] = useState<SelectProps.Option>(widget.options[0]);
+  const [selectedOption, setSelectedOption] = useState<SelectProps.Option | null>(widget.options[0]);
+  const options = widget.options.map(({ label }) => ({ label, value: label }));
+  const disabled = !readOnly || options.length === 0;
+
+  const changeOption: NonCancelableEventHandler<SelectProps.ChangeDetail> = ({ detail: { selectedOption } }) => {
+    setSelectedOption(selectedOption);
+  };
+
+  useEffect(() => {
+    if (options.length === 0) {
+      setSelectedOption(null);
+    } else if (!selectedOption || options.filter(({ label }) => selectedOption.label === label).length === 0) {
+      setSelectedOption(widget.options[0]);
+    }
+  }, [widget.options]);
 
   return (
     <SpaceBetween size={'xs'} direction={'horizontal'}>
```

Comment:
avoid `{'string'}`, just "string" instead.

Reply:
This is a pattern we have been following all over the app. Maybe we want to include a prettier/eslint config enforcing this. I would say this is out of the scope of this PR but worth tracking.

Reply:
yes we should update the eslint config

Reply:
created separate PR for this https://github.com/awslabs/iot-app-kit/pull/584